### DISCUSSION TITLE
Using access_token from SSO user in a custom endpoint

### DISCUSSION THREAD

**Discussion by nicolaschapeau** [directus/directus#12262] [Archive]
Hello,

I've succesfully integrated SSO with a custom provider.
My goal would be to use the provider API to execute actions on their end.
To do so, I would need the user's access_token to authorize my requests.

I see there is a auth_data that we can retrieve, but there is only a refresh_token in it. 
And i not sure if it's a good idea to use it to get an access_token.

I can't see any way in directus code to get this access_token. Is there a way ?
Or should I implement a sort of custom code to refresh and get an access_token ?

Regards, Nicolas.

**Comment by rijkvanzanten** [comment]
There's currently no way to access / use the access token from the third party auth provider when logging in, so you'll have to rely on some custom logic to refresh yourself using the stored refresh token 👍🏻

**Reply by nicolaschapeau** [reply]
Okay thanks !
I'll do that, could be useful to implement it later on
